### PR TITLE
feat: Creates a modal showing all the available keyboard shortcuts

### LINKED ISSUE #2717: feat: Modal showing all the available keyboard shortcuts
- Define a shortcut to list all keyboard shortcuts
- show all the keyboard shortcuts in a modal
- Define UI element which can be clicked to view the keyboard shortcuts

### PR DESCRIPTION
**Pull request by iamsivin**
Fixes #2717 

https://user-images.githubusercontent.com/64252451/128690095-d05568e4-b91d-4040-be56-321afdb3fdd0.mov

### CODE REVIEW

File: app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+                  C
+                </span>
+              </div>
+            </div>
+            <div class="text-key">
+              <h6 class="sub-block-title">
+                Go to Contacts Dashboard
+              </h6>
+              <div class="shortcut-key__wrap">
+                <span class="shortcut-key">
```
Comment (pranavrajs) [inline app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue:201]: Please use i18n in this file.

File: app/javascript/dashboard/components/layout/Sidebar.vue [hunk 2]
```diff
@@ -86,6 +92,8 @@ import OptionsMenu from './sidebarComponents/OptionsMenu.vue';
 import AccountSelector from './sidebarComponents/AccountSelector.vue';
 import AddAccountModal from './sidebarComponents/AddAccountModal.vue';
 import AddLabelModal from '../../routes/dashboard/settings/labels/AddLabel';
+import wootKeyShortcutModal from 'components/widgets/modal/WootKeyShortcutModal';
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/layout/Sidebar.vue:95]:
```suggestion
import WootKeyShortcutModal from 'components/widgets/modal/WootKeyShortcutModal';
```

File: app/javascript/dashboard/components/layout/Sidebar.vue [hunk 3]
```diff
@@ -253,8 +263,24 @@ export default {
     this.$store.dispatch('notifications/unReadCount');
     this.$store.dispatch('teams/get');
     this.setChatwootUser();
+    document.addEventListener('keydown', this.handleKeyEvents);
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/layout/Sidebar.vue:266]: Use `eventListene` mixin after merging the PR https://github.com/chatwoot/chatwoot/pull/2672

File: app/javascript/dashboard/components/layout/Sidebar.vue [hunk 4]
```diff
@@ -253,8 +263,24 @@ export default {
...
+  },
+
+  destroyed() {
+    document.removeEventListener('keydown', this.handleKeyEvents);
   },
   methods: {
+    toggleKeyShortcutModal() {
+      this.showShortcutModal = !this.showShortcutModal;
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/layout/Sidebar.vue:274]: I think `this.showShortcutModal  = true` is enough

File: app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </transition>
+</template>
+
+<script>
+const SHORTCUT_KEYS = [
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue:85]: Move this to a separate constant file for better readability

File: app/javascript/shared/helpers/KeyboardHelpers.js [hunk 6]
```diff
@@ -9,3 +9,7 @@ export const isEscape = e => {
 export const hasPressedShift = e => {
   return e.shiftKey;
 };
+
+export const hasPressedCommandAndForwardslash = e => {
```
Comment (muhsin-k) [inline app/javascript/shared/helpers/KeyboardHelpers.js:13]:
```suggestion
export const hasPressedCommandAndForwardSlash = e => {
```

**Review by muhsin-k — CHANGES_REQUESTED** [review]
@iamsivin Great work. Please see the inline comments

**Comment by muhsin-k** [comment]
@iamsivin Modal is not closing after clicking outside of modal

**Review by qamar37 — COMMENTED** [review]
😄😄😄

File: app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+                {{ $t('KEYBOARD_SHORTCUTS.KEYS.ALT_OR_OPTION_KEY') }}
+              </span>
+              <span class="shortcut-key key">
+                K
+              </span>
+            </div>
+          </div>
+
+          <div class="title-key__wrap">
+            <h6 class="sub-block-title">
```
Comment (nithindavid) [inline app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue:49]: Use `p` ans `span` and `font-size-small`

File: app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+  width: fit-content;
+}
+
+.page-top-bar {
+  padding: var(--space-medium) var(--space-large) var(--space-large)
+    var(--space-large);
+}
+
+.title-wrap {
+  display: flex;
```
Comment (nithindavid) [inline app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue:117]: Can we check if `display: flex` is necessary here and also for css written below ?

**Review by nithindavid — CHANGES_REQUESTED** [review]
ALso please see

File: app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue (hunk 8 above)
Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/modal/WootKeyShortcutModal.vue:117]: Yes, I have checked, I think we needed `display: flex` here.

**Review by nithindavid — APPROVED** [review]

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**